### ISSUE TITLE
llvm: No such file or directory

### ISSUE THREAD

**Issue by hsarmadi** [mihaibujanca/dynamicfusion#54]
Hi! Thank you for your work. We are trying to use your code to do an experiment for our paper. We managed to compile in Ubuntu 16.04 using your build script and we are running it on the default test data (umbrella) however we get the following error:

**llvm: No such file or directory**

We debugged the error up to this point:

OptSolver.h: 50: m_optimizerState = Opt_NewState(initParams);

But it is not possible to debug further because it is calling code that seemingly is compiled by terra with no debug symbols. We tried both gpu_optimized and master branch but we get the same error.

We would appreciate your help

**Comment by mihaibujanca** [comment]
Had this error in the past but haven't been able to reproduce it lately. Will give it a try in the upcoming days

**Comment by chenkl1995** [comment]
I try to compile in Windows with VS2013, Cuda 8.0.
And I also meet the debug error in this point:

    OptSolver.h: 50: m_optimizerState = Opt_NewState(initParams);
And if I directly run the `dynamicfusiond.exe`, the cmd shows:
    ![image](https://user-images.githubusercontent.com/15438055/39738357-3cf195c4-52bd-11e8-8c6b-2d05aa5dcced.png)
I guess the problem maybe in Opt, and I found the same question with me in [niessner/Opt issuse #64](https://github.com/niessner/Opt/issues/64), the issue closed with changing windows to ubuntu, do you have better suggestions?
<br/>
Thanks.

**Comment by jhz-linhe** [comment]
I encountered the same prolbem [llvm: No such file or directory] in ubuntu16.04, compiled successfully. Thanks for any update.

**Comment by YJonmo** [comment]
Hi there, I got the same error. It only runs for the first image but for the second image, it gives the llvm error. 

Ubuntu 16.04 here.

**Comment by YJonmo** [comment]
I finally got it working by downgrading CUDA to 8 from 9.

Now the sample code for Umbrella is running already an hour and is not finished yet. Is it supposed to be this slow or is there anything wrong?

**Comment by caoPhoenix** [comment]
Yeah, it's useful to update my CUDA 7.5 to CUDA 8.0. And now the system is running with data Umbrella but very very slow. My GPU is GTX960M. Is this speed normal?

**Comment by caoPhoenix** [comment]
Well, I meet a new problem when I want to build this system again. That is:

`[  6%] Building NVCC (Device) object kfusion/CMakeFiles/kfusion.dir/src/cuda/kfusion_generated_proj_icp.cu.o
[ 18%] Building NVCC (Device) object kfusion/CMakeFiles/kfusion.dir/src/cuda/kfusion_generated_imgproc.cu.o
[ 18%] Building NVCC (Device) object kfusion/CMakeFiles/kfusion.dir/src/cuda/kfusion_generated_tsdf_volume.cu.o
CMake Error at kfusion_generated_proj_icp.cu.o.cmake:207 (message):
  Error generating
  /home/ctcer/dynamicfusion/build/kfusion/CMakeFiles/kfusion.dir/src/cuda/./kfusion_generated_proj_icp.cu.o


kfusion/CMakeFiles/kfusion.dir/build.make:752: recipe for target 'kfusion/CMakeFiles/kfusion.dir/src/cuda/kfusion_generated_proj_icp.cu.o' failed
make[2]: *** [kfusion/CMakeFiles/kfusion.dir/src/cuda/kfusion_generated_proj_icp.cu.o] Error 1
make[2]: *** 正在等待未完成的任务....
CMake Error at kfusion_generated_imgproc.cu.o.cmake:207 (message):
  Error generating
  /home/ctcer/dynamicfusion/build/kfusion/CMakeFiles/kfusion.dir/src/cuda/./kfusion_generated_imgproc.cu.o


kfusion/CMakeFiles/kfusion.dir/build.make:520: recipe for target 'kfusion/CMakeFiles/kfusion.dir/src/cuda/kfusion_generated_imgproc.cu.o' failed
make[2]: *** [kfusion/CMakeFiles/kfusion.dir/src/cuda/kfusion_generated_imgproc.cu.o] Error 1
CMake Error at kfusion_generated_tsdf_volume.cu.o.cmake:207 (message):
  Error generating
  /home/ctcer/dynamicfusion/build/kfusion/CMakeFiles/kfusion.dir/src/cuda/./kfusion_generated_tsdf_volume.cu.o


kfusion/CMakeFiles/kfusion.dir/build.make:289: recipe for target 'kfusion/CMakeFiles/kfusion.dir/src/cuda/kfusion_generated_tsdf_volume.cu.o' failed
make[2]: *** [kfusion/CMakeFiles/kfusion.dir/src/cuda/kfusion_generated_tsdf_volume.cu.o] Error 1
CMakeFiles/Makefile2:85: recipe for target 'kfusion/CMakeFiles/kfusion.dir/all' failed
make[1]: *** [kfusion/CMakeFiles/kfusion.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
`

**Comment by caoPhoenix** [comment]
The new problem above was solved by reinstall :
sudo apt-get install nvidia-cuda-dev nvidia-cuda-toolkit nvidia-nsight nvidia-prime
The reason should be that I uninstall cuda while I install cuda 8.0.(sudo apt-get autoremove nvidia-cuda-toolkit)
And now when I type "nvcc -V" it show cuda 7.5 but the system run without "llvm:....".

**Comment by zj19941113** [comment]
> The new problem above was solved by reinstall :
> sudo apt-get install nvidia-cuda-dev nvidia-cuda-toolkit nvidia-nsight nvidia-prime
> The reason should be that I uninstall cuda while I install cuda 8.0.(sudo apt-get autoremove nvidia-cuda-toolkit)
> And now when I type "nvcc -V" it show cuda 7.5 but the system run without "llvm:....".

thx！

**Comment by tangweiyill** [comment]
https://github.com/niessner/Opt/issues/143   and a new terra may solve this problem
